controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -321,19 +322,18 @@ private ApplicationController applications() {
     /** Finds the next step to trigger for the given application, if any, and returns these as a list. */
     private List<Job> computeReadyJobs(DeploymentStatus status) {
         List<Job> jobs = new ArrayList<>();
-        status.jobsToRun().forEach((job, versionsList) -> {
-            for (Versions versions : versionsList)
-                status.jobSteps().get(job).readyAt(status.application().require(job.application().instance()).change())
-                      .filter(readyAt -> ! clock.instant().isBefore(readyAt))
-                      .filter(__ -> ! (job.type().isProduction() && isUnhealthyInAnotherZone(status.application(), job)))
-                      .filter(__ -> abortIfRunning(versionsList, status.jobs().get(job).get())) // Abort and trigger this later if running with outdated parameters.
-                      .ifPresent(readyAt -> {
-                          jobs.add(deploymentJob(status.application().require(job.application().instance()),
-                                                 versions,
-                                                 job.type(),
-                                                 status.instanceJobs(job.application().instance()).get(job.type()),
-                                                 readyAt));
-                      });
+        Map<JobId, List<DeploymentStatus.Job>> jobsToRun = status.jobsToRun();
+        jobsToRun.forEach((job, versionsList) -> {
+            versionsList.get(0).readyAt()
```

Comment:
Better to store the readyAt where we compute the job versions, and have all data available.